### PR TITLE
Use simple Rust code to specify map importing config, instead of conf…

### PR DESCRIPTION
**Pull request by dabreegster**
…ig files.

Any feedback on this approach, @mdejean, since you recently touched this code? I think it would've been much simpler if things had been structured this way.

# Problems

Brief history lesson: Originally the importer just worked for Seattle. Then we hardcoded a few more cities in Rust. As things grew, I split out config files, thinking this would make the process easier for newcomers using the CLI. But eventually importing a map became possible through the UI without even having to create a config file, manually figure out the Geofabrik OSM source to use, etc.

I think we've outgrown the config files. They're tedious to maintain when we introduce new options. Most cities use default settings, but when one place has an override for a good reason, other places accidentally copy that later, and keeping track is confusing.

# Solution

Get rid of the config files and just have a single function that generates the `convert_osm::Options` using the name of the map. Use conditionals and normal programming constructs to express edge cases.

I thought about YAML or some kind of config that would let a city base its settings off a template and override some stuff. I'm not convinced that's any better than a small bit of Rust. Related reading: https://mikehadlow.blogspot.com/2012/05/configuration-complexity-clock.html, https://beepb00p.xyz/configs-suck.html

# The result

I think the code is much simpler to follow. Seattle is less of a special case. We rely on convention over configuration in one place (specifying extra residential buildings from a JSON file named a certain way).

# Next steps

We store a few redundant copies of some osm.pbf files, from cities that all exist near each other. I'm starting to think about moving those huge input files into `data/input/shared`, like we do for elevation data. I want to avoid making someone download all of them though, which happens right now for anyone using _any_ elevation data.

### CODE REVIEW

File: importer/src/map_config.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+use abstio::{CityName, MapName};
+use geom::Distance;
+use map_model::DrivingSide;
+
+/// Given the name of a map, configure its import.
```
Comment (dabreegster) [inline importer/src/map_config.rs:5]: All of those deleted config files are now generated by this one bit of code. I feel like this is way simpler

File: importer/src/map_config.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            } else {
+                Distance::meters(8.0)
+            },
+            turn_on_red: name.city.country == "us" && name.city.city != "nyc",
+        },
+        onstreet_parking: match name.city.city.as_ref() {
+            "seattle" => {
+                convert_osm::OnstreetParking::Blockface(name.city.input_path("blockface.bin"))
+            }
+            "lyon" | "milwaukee" | "montreal" | "tel_aviv" | "zurich" => {
```
Comment (dabreegster) [inline importer/src/map_config.rs:38]: This is an example where people probably copied an override around. I haven't tracked down the original place that used this.

File: importer/src/map_config.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    "downtown" => 5,
+                    "lakeslice" => 5,
+                    "qa" => 5,
+                    "south_seattle" => 5,
+                    "wallingford" => 5,
+                    _ => 1,
+                },
+            )
+        } else {
+            convert_osm::PrivateOffstreetParking::FixedPerBldg(3)
```
Comment (dabreegster) [inline importer/src/map_config.rs:64]: A bunch of places used 5 or 10 instead, which are all arbitrary. Just setting this to 3. This assigns all buildings 3 parking spots that're physically inside the building (a garage or car-port) or around the back or in the driveway or something. Very low-quality guess, but we haven't had time to develop any better heuristics or pull in other data sources for anywhere except Seattle.

File: importer/src/utils.rs [hunk 4]
```diff
@@ -111,10 +112,53 @@ pub fn osmconvert(
...
+pub async fn osm_to_raw(
+    name: MapName,
+    timer: &mut abstutil::Timer<'_>,
+    config: &ImporterConfiguration,
+) -> RawMap {
+    let boundary_polygon = format!(
+        "importer/config/{}/{}/{}.poly",
+        name.city.country, name.city.city, name.map
+    );
+    let osm_url = crate::pick_geofabrik(boundary_polygon.clone())
```
Comment (dabreegster) [inline importer/src/utils.rs:125]: Before, we manually specified geofabrik URLs, which were found using this tool. Let's skip a step and just always run that tool to get the source URL.

File: map_model/src/make/traffic_signals/mod.rs [hunk 5]
```diff
@@ -9,7 +9,7 @@ use std::collections::{BTreeSet, HashSet};
 
 use crate::{
     ControlTrafficSignal, DrivingSide, Intersection, IntersectionCluster, IntersectionID, Map,
-    MovementID, RoadID, Stage, StageType, TurnPriority, TurnType, MapConfig
+    MapConfig, MovementID, RoadID, Stage, StageType, TurnPriority, TurnType,
```
Comment (dabreegster) [inline map_model/src/make/traffic_signals/mod.rs:12]: This just looks like `cargo fmt`

**Review by dabreegster — COMMENTED** [review]
I'm regenerating all maps. I expect some diffs, since I got rid of all the overrides for offstreet parking. I'll check carefully.

**Comment by Robinlovelace** [comment]
image

Win in terms of codebase size and maintainability!

**Comment by dabreegster** [comment]
I'm convinced this is the way forward, so merging, but async feedback always welcome

**Pull request merged**